test.sh

```diff
@@ -23,29 +23,77 @@ set -euo pipefail
 
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 TEMP_DIR=$(mktemp -d -t crd-ref-docs-XXXXX)
-ACTUAL="${TEMP_DIR}/out.asciidoc"
-DEFAULT_ARGS=(--log-level=ERROR --source-path="${SCRIPT_DIR}/test" --renderer=asciidoctor --output-path="${ACTUAL}")
+DEFAULT_ARGS=(--log-level=ERROR --source-path="${SCRIPT_DIR}/test" --output-path="${TEMP_DIR}/out")
 
 trap '[[ $TEMP_DIR ]] && rm -rf "$TEMP_DIR"' EXIT
 
 run_test() {
-    ARGS=("${DEFAULT_ARGS[@]}" "$@")
-    rm -f "$ACTUAL"
+    local actual="${TEMP_DIR}/out"
+    rm -f "$actual"
+
+    local renderer=asciidoctor
+    local templates_dir=
+
+    while :; do
+        case "${1:-}" in
+            --renderer)
+                if [[ -n "${2:-}" ]]; then
+                    renderer="$2"
+                    shift
+                else
+                    printf "ERROR: '--renderer' cannot be empty.\n\n" >&2
+                    exit 1
+                fi
+                ;;
+            --templates-dir)
+                if [[ -n "${2:-}" ]]; then
+                    templates_dir="$2"
+                    shift
+                else
+                    printf "ERROR: '--templates-dir' cannot be empty.\n\n" >&2
+                    exit 1
+                fi
+                ;;
+            *)
+                break
+                ;;
+        esac
+
+        shift
+    done
+
+    local args=("${DEFAULT_ARGS[@]}" --renderer="$renderer")
+    if [[ -n "$templates_dir" ]]; then
+        args+=(--templates-dir="$templates_dir")
+    fi
+
+    local expected
+    if [[ "$renderer" == "asciidoctor" ]]; then
+        expected=expected.asciidoc
+    else
+        expected=expected.md
+    fi
 
     (
         cd "$SCRIPT_DIR"
-        go run main.go "${ARGS[@]}"
-        DIFF=$(diff -a -y --suppress-common-lines "${SCRIPT_DIR}/test/expected.asciidoc" "$ACTUAL") || true
-        if [ "$DIFF" ]; then
+        cmd=(go run main.go "${args[@]}")
+        echo "${cmd[@]}"
+
+        "${cmd[@]}"
```

Comment:
Nit: IIUC we want to print the full command that we are testing. I would have just repeated the command with an echo.
```suggestion
         echo "go run main.go ${args[@]}"
         go run main.go "${args[@]}"
```

Reply:
Why do you like that better? Using a variable, you'd only have to do any changes in one place.

Reply:
For no real good reason, I used to avoid using bash arrays. Your argument makes sense. I'm fine with your proposal.